### PR TITLE
Refactors ssh_config resource and minor updates and fixes in docs

### PR DESCRIPTION
**Pull request by luispresuelVenafi**
Refactors `ssh_config` resource and flow in order to make the access_token mandatory from the provider which always will return the principals from the template. Updates documentation accordingly. Fixes documentation fro `venafi_ssh_certificate` resource. Fixes typo in message name for the logs.

### CODE REVIEW

**Review by ricrodriguezg — APPROVED** [review]

File: venafi/provider.go [hunk 1]
```diff
@@ -12,7 +12,7 @@ import (
 
 const (
 	messageVenafiPingFailed       = "Failed to ping Venafi endpoint: "
-	messageVenafiPingSucessfull   = "Venafi ping sucessfull"
+	messageVenafiPingSucessful    = "Venafi ping sucessful"
```
Comment (angelmoo) [inline venafi/provider.go]: I think is: successful

File: venafi/resource_venafi_ssh_certificate.go [hunk 2]
```diff
@@ -97,11 +97,21 @@ func resourceVenafiSshCertificate() *schema.Resource {
...
-				Elem:        &schema.Schema{Type: schema.TypeString},
+				Type:          schema.TypeList,
+				Description:   "The requested principals.",
+				ForceNew:      true,
+				Optional:      true,
+				Elem:          &schema.Schema{Type: schema.TypeString},
+				ConflictsWith: []string{"principals"},
+				Deprecated:    "This will be removed in the future. Use \"principals\" instead",
+			},
+			"principals": &schema.Schema{
```
Comment (angelmoo) [inline venafi/resource_venafi_ssh_certificate.go:108]: is there a code change to use this principals instead of principal?

**Review by angelmoo — CHANGES_REQUESTED** [review]
just some change requests

File: venafi/resource_venafi_ssh_certificate_test.go [hunk 3]
```diff
@@ -60,7 +96,87 @@ func TestSshCert(t *testing.T) {
...
+}
+
+func TestSshCertLocalPublicKey(t *testing.T) {
+	t.Log("Testing creating ssh certificate")
+
+	data := getTestData()
+	data.publicKeyMethod = "local"
+
+	// data.principals only holds the values for principals, actually we are testing "principal" attribute defined at the resource.
+	config := fmt.Sprintf(tppSshCertResourceTest, tokenSshCertProv, data.keyId, data.template, data.publicKeyMethod, data.validityPeriod, data.principals, data.sourceAddress)
```
Comment (angelmoo) [inline venafi/resource_venafi_ssh_certificate_test.go:153]: Could we use the new resources that works with principals on this test?

**Review by angelmoo — CHANGES_REQUESTED** [review]

File: venafi/resource_venafi_ssh_certificate_test.go (hunk 3 above)
Comment (luispresuelVenafi) [reply venafi/resource_venafi_ssh_certificate_test.go:153]: We are covering that use case in the test ` TestSshCertLocalPublicKeyNewAttrPrincipals`

File: venafi/resource_venafi_ssh_certificate.go (hunk 2 above)
Comment (luispresuelVenafi) [reply venafi/resource_venafi_ssh_certificate.go:108]: Fixed it.

**Review by angelmoo — APPROVED** [review]
code changes looks good

**Pull request merged**